### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by zapaiamarce** [Alvax10/dwf-m10#2]
Hola! Te paso mi feedback a nivel código:

Sobre el nombre de las carpetas de los componentes: Usá la convención que quieras pero usá una sola:
![image](https://user-images.githubusercontent.com/1208547/174883678-41270176-dba9-42b5-b76d-d6882f77b1c3.png)
Creo que la mejor es usar guiones-medios-para-los-nombres y no mayúsculas. Sobre todo para nombres de carpetas y archivos es lo mejor.

La mayoría de los componentes no están formateados, se ven asi:
![image](https://user-images.githubusercontent.com/1208547/174883956-ee283d94-cf2f-4246-9c04-ed1c5db1c8b8.png)
Y si los formateas se ven ordenados:
![image](https://user-images.githubusercontent.com/1208547/174884011-af9a213d-f988-4f84-8ee0-18a45d6e49aa.png)
Esto es super importante. Todo tiene que estar formateado para poder ser revisado.
Guarda con usar sub carpetas para sub componentes:
![image](https://user-images.githubusercontent.com/1208547/174884197-c80892e2-3461-4b1e-a5c2-112af4e68ad7.png)
Esto después se te puede descontrolar.
Acá usas un div con un onClick para ir a una ruta?
![image](https://user-images.githubusercontent.com/1208547/174884304-90795678-1bd4-452f-8ba7-25f59ed10d74.png)
Podés usar el elemento <a> o usar un Link de next https://nextjs.org/docs/api-reference/next/link

Acá usas un useEffect para algo que podés resolver sin usar ningun hook
![image](https://user-images.githubusercontent.com/1208547/174884487-9aad6004-9647-4d74-baa4-c25e3c1ed8b6.png)
Asi podés hacerlo sin usar hooks de más:
![image](https://user-images.githubusercontent.com/1208547/174884648-961852d1-bfa4-4f0a-a782-31082c08cd35.png)
Guarda con usar hooks para cosas que no son necesarias.
El timeOut?
![image](https://user-images.githubusercontent.com/1208547/174884843-13beeb98-1e5e-462b-b1ec-e0ed08468055.png)

Esta práctica de extender tu componente con un styled y maquetar todo con clases no está mal pero te va a traer problemas:
![image](https://user-images.githubusercontent.com/1208547/174885238-c32f9789-6bab-43c1-aa1c-90ae1ba9a4d8.png)
Si bien lo mejor es crear un styled component para cada ladrillo si seguís con la linea de tu técnica quizás lo mejor es que crees un styled que envuelva a todo tu componente:
![image](https://user-images.githubusercontent.com/1208547/174885389-eae30716-d13c-40ad-9205-b7cd01441da8.png)

Y usarlo asi:
![image](https://user-images.githubusercontent.com/1208547/174885500-96e08fde-51c0-4ad0-8ee2-a236b345f97a.png)

Maquetar directamente con props no es la mejor idea:
![image](https://user-images.githubusercontent.com/1208547/174885610-e1ab86be-3a27-4694-8673-4c1167ebce24.png)
Para esto lo mejor es extender el botón principal y sumarle lo que necesites:
![image](https://user-images.githubusercontent.com/1208547/174885925-abf9d6ab-5b6c-46ec-ad17-b05e26bd7d56.png)

Para mi a nivel código hace falta una revisada general y minuciosa, siendo el último módulo técnico de la carrera me gustaría que tu código esté impecable asi que metele todo el tiempo que necesites, empezando con los básicos como formatear todo y después revisar cada componente y tratando de llevarlo al máximo de prolijidad.

Hablando de la experiencia, cuando pongo comprar en un producto me lleva acá y listo:
![image](https://user-images.githubusercontent.com/1208547/174886321-df292054-be85-4f12-872d-9038ba637030.png)

Debería chequear si tengo sesión y si no tengo debería llevarme a iniciar sesión, una vez que tenga la sesión, debería llevarme a pagar y asi todo el flujo normal de una compra. Si hace falta más detalles en el desafío lo vemos en discord, pero creo el flujo básico se entiende.

Yo lo veo super encaminado pero hay que llevarlo a otro grado de prolijidad a nivel código y funcional. Esto te va a ayudar mucho a poner el foco en los detalles y a llevarte de la carrera no solo aprendizajes técnicos.

**Comment by Alvax10** [comment]
Marce, el componente productRoot de la forma que lo estás usando ahí ( o por lo menos de la forma que lo entiendo ) es que del styled lo pasas al componente de "productId" y de ahí manipulas la l+ogica con el componente. Pero de la forma que lo estoy usando yo, el componente usa la lógica, el styled le da los estilos y la page "/products/[id]" lo muestra, para hacerlo de la forma que me decís, no tendría que cambiar toda esa lógica?

**Comment by zapaiamarce** [comment]
Claro, lo que vos hacés es usar un styled para inyectarle una mini hoja de estilos. La diferencia no es tan grande pero en tu implementación la jerarquía es:

Page >  Styled (del component) > Component

No me parece que esté mal técnicamente hablando, pero si venimos hablando de otro tipo de jerarquías donde 

Page > Componente > UI component

El ui component es un "ladrillo" que el componente usa y puede ser propio o de la capa UI. El tema de saltearse la capa y hacer un styled que envuelva al componente original es que te pone el styled en el medio de toda la cuestión. Desde fuera del componente solo debería haber un import del componente y manipular lo que suceda vía props. Repito, técnicamente lo que hacés no está mal pero tenemos que tratar de respetar las jerarquías que proponemos siempre para que toda tu estructura sea homogénea y por ende predecible.
